src/olympia/access/acl.py

```diff
@@ -22,16 +22,13 @@ def action_allowed(request, app, action):
     'Admin:%' is true if the user has any of:
     ('Admin:*', 'Admin:%s'%whatever, '*:*',) as rules.
     """
-    allowed = any(match_rules(group.rules, app, action) for group in
-                  getattr(request, 'groups', ()))
-    return allowed
+    return action_allowed_user(request.user, app, action)
 
 
 def action_allowed_user(user, app, action):
     """Similar to action_allowed, but takes user instead of request."""
-    allowed = any(match_rules(group.rules, app, action) for group in
-                  user.groups.all())
-    return allowed
+    return (user.is_authenticated() and any(
+        match_rules(group.rules, app, action) for group in user.groups_list))
```

Comment:
I feel like something like 

```python
if not user.is_authenticated():
    return

matches = (match_rules(group.rules, app, action) for group in user.groups_list)
return any(matches)
```

makes it more obvious about the fact that we're not evaluating anything if the user is not authenticated and looks better imho, but that's just a nitpick.